volumina/pixelpipeline/datasources/cachesource.py

```diff
@@ -0,0 +1,120 @@
+import threading
+import sys
+
+from PyQt5.QtCore import QObject, pyqtSignal
+
+from volumina.pixelpipeline.interface import DataSourceABC
+from volumina.slicingtools import is_pure_slicing
+from volumina.utility.cache import KVCache
+from volumina.config import CONFIG
+
+
+ARRAY_CACHE = KVCache(CONFIG.cache_size, getsizeof=sys.getsizeof)
```

Comment:
are we sure we'll never get a view as data? in that case `sys.getsizeof` won't work.

```ipython
In [2]: x = numpy.ones((10000, 2000))                                                                                                      

In [3]: sys.getsizeof(x)                                                                                                                   
Out[3]: 160000112

In [4]: y = x.reshape((10000, 2000))                                                                                                       

In [5]: sys.getsizeof(y)                                                                                                                   
Out[5]: 112

In [6]: x.nbytes                                                                                                                           
Out[6]: 160000000

In [7]: y.nbytes                                                                                                                           
Out[7]: 160000000
```

or would you say that is not a problem?

The problem I see is that even if I delete `x`, `getsizeof(y)` will not return the true number